src/types/Patterns.ts

```diff
@@ -18,9 +18,30 @@
  * SPDX-License-Identifier: Apache-2.0
  ********************************************************************************/
 
+import { getRequireHttpsUrlPattern } from 'services/EnvironmentService'
+
+// check the REQUIRE_HTTPS_URL_PATTERN environment variable, defaulting to 'true' if not set
+const requireHttpsUrlPattern = getRequireHttpsUrlPattern() === 'true'
+if (requireHttpsUrlPattern) {
+  console.log('HTTPS in URL patterns is required.')
+} else {
+  console.log('HTTPS in URL patterns is not required.')
+}
+
 const DOMAIN =
   /([a-z0-9]|[a-z0-9][a-z0-9-]{0,61}[a-z0-9])(\.([a-z0-9]|[a-z0-9][a-z0-9-]{0,61}[a-z0-9])){1,10}/i
 const URLPATH = /(\/[a-z0-9-._~:/?#[\]@!$&'()*+,;=%]{0,500}){0,20}/
+// construct regex patterns for URL based on the REQUIRE_HTTPS_URL_PATTERN environment variable
+const urlPattern = requireHttpsUrlPattern
+  ? new RegExp(
+      `^(https)://(${DOMAIN.source})(:\\d{1,5})?(${URLPATH.source})?$`,
+      'i'
+    )
+  : new RegExp(
+      `^(https?)://(${DOMAIN.source})(:\\d{1,5})?(${URLPATH.source})?$`,
+      'i'
+    )
```

Comment:
This can be written with less duplication as:
`^(https${requireHttpsUrlPattern ? '' : '?'})://(${DOMAIN.source})(:\\d{1,5})?(${URLPATH.source})?$`
or for better readability:
`^(http${requireHttpsUrlPattern ? 's' : 's?'})://etc...`